compiler/noirc_evaluator/src/ssa/ir/dfg.rs

```diff
@@ -160,7 +160,7 @@ impl DataFlowGraph {
         call_stack: CallStack,
     ) -> InsertInstructionResult {
         use InsertInstructionResult::*;
-        match instruction.simplify(self, block, ctrl_typevars.clone()) {
+        match instruction.simplify(self, block, ctrl_typevars.clone(), &call_stack) {
```

Comment:
I'm not a fan of adding this new argument but it seems like the cleanest way for now.

Reply:
We can explore a `SimplifiedToTwoInstructions` or an arbitrary size variant in the future. IIRC I've been down that path before as well though and it caused other issues then.